lib/std/thread.zig

```diff
@@ -500,4 +500,33 @@ pub const Thread = struct {
         };
         return @intCast(usize, count);
     }
+
+    pub fn getCurrentThreadId() u64 {
+        switch (std.Target.current.os.tag) {
+            .linux => {
+                // Use the syscall directly as musl doesn't provide a wrapper.
+                return @bitCast(u32, os.linux.gettid());
+            },
+            .windows => {
+                return os.windows.kernel32.GetCurrentThreadId();
+            },
+            .macos, .ios, .watchos, .tvos => {
+                var thread_id: u64 = undefined;
+                // Pass thread=null to get the current thread ID.
+                assert(c.pthread_threadid_np(null, &thread_id) == 0);
+            },
```

Comment:
this prong does not return, is that an oversight?

Reply:
Good catch!